docs/articles/vector_search_based_recommender_system.md

```diff
@@ -0,0 +1,276 @@
+
+![overview](../assets/use_cases/vector_search_based_recommender_system/image3.png)
+
+## Steps to build the Recommender System using Rockset and OpenAI Embedding
+
+Let's begin with signing up for Rockset and OpenAI and then dive into all the steps involved within the Google Colab notebook to build our recommendation webapp:
+
+### Step 1: Initiate your vector database
+
+In this example we are using [Rockset](https://rockset.com/create/) as a vector database. Once signed up, create an [API key](https://console.rockset.com/apikeys) to use in the backend code:
+
+```python
+import os
+os.environ["ROCKSET_API_KEY"] = "XveaN8L9mUFgaOkffpv6tX6VSPHz####"
+```
+
+### Step 2: Create a new Collection and Upload Data
+
+For this tutorial, we'll be using Amazon product review [data](https://drive.google.com/file/d/1EEvCUqKIH6LuLLGRjo6ayCEXqT4Z-ZVE/view?usp=drive_link). Download this on your local machine so it can be uploaded to your collection.
+
+We'll be uploading the [sample_data.json](https://drive.google.com/file/d/1EEvCUqKIH6LuLLGRjo6ayCEXqT4Z-ZVE/view?usp=drive_link) file to Rockset.
+
+Note: In practice, the data is usually ingested from another streaming service but for keeping things simple and building a demo application we are using a sample from a public dataset.
+
+### Step 3: Create OpenAI API Key
+
+To convert data into [embeddings](https://platform.openai.com/docs/api-reference/embeddings/create), we'll use OpenAI's model.
+
+After signing up, go to [API Keys](https://platform.openai.com/api-keys) and create a secret key. Don't forget to copy and save your key that will look similar to "sk-***********************". Like Rockset's API key, save your OpenAI key in the environment so it can easily used throughout the code:
+
+```python
+import os
+os.environ["OPENAI_API_KEY"] = "sk-####"
+```
+### Step 4: Create a Query Lambda on Rockset
+
+[Query Lambdas](https://docs.rockset.com/documentation/docs/query-lambdas) are named, parameterized SQL queries stored in Rockset that can be executed from a dedicated REST endpoint. Using Query Lambdas, you can save your SQL queries as separate resources and manage them successfully through development and production.
+
+Let's create one for our tutorial. We'll be using the following Query Lambda with parameters: embedding, brand, min_price, max_price and limit.
+
+```sql
+SELECT
+  asin,
+  title,
+  brand,
+  description,
+  estimated_price,
+  brand_tokens,
+  image_ur1,
```

Comment:
presumably, this should be "image_url"?

Reply:
@fatirmalik1
presumably, this should be "image_url"? same question below (line 80).

Reply:
the sample data has a schema with that column name, "image_ur1" or url.

Reply:
Just to be really clear, ur and then the number one?
url or ur1 ?

Reply:
@@ankit1khare
To be absolutely clear: you mean
ur and then the number one (1) = ur1
OR
ur and then the letter l = url

Please clarify.

Reply:
yes, let's make it url at all the three places plz